### PR TITLE
Fix issues in tests for cache mediator

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/CacheTestCase.java [hunk 1]
```diff
@@ -68,38 +67,32 @@ public void close() throws Exception {
     }
 
     @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
-    @Test(groups = {"wso2.esb"}, description = "Tests whether maxSize value evicts the new messages")
+    @Test(groups = {"wso2.esb"}, description = "Tests whether maxSize value evicts the older messages")
     public void testMaxSize() throws AxisFault, XPathExpressionException, InterruptedException {
         OMElement response;
 
-        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "ABCD");
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxSize"), "", "ABCD");
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/CacheTestCase.java]: Shall we define a constant for "maxSize"?

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/CacheTestCase.java [hunk 2]
```diff
@@ -68,38 +68,33 @@ public void close() throws Exception {
...
+        assertEquals(responseABC1, responseABC2, "The response for ABC is cached");
 
-        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "ABCD");
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL(apiName), "", "ABCD");
         String responseABCD2 = response.getFirstElement().toString();
 
-        if (!responseABCD1.equalsIgnoreCase(responseABCD2)) {
-            assertTrue(true, "The response ABCD has been evicted");
-        }
+        assertNotEquals(responseABCD1, responseABCD2, "The response for ABCD has been evicted");
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/CacheTestCase.java]: Is the message printed when the assertion fails, correct?

Comment (riyafa) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/CacheTestCase.java]: Fixed in #1167

File: integration/mediation-tests/tests-sample/src/test/java/org/wso2/carbon/esb/samples/test/mediation/cache/Sample420TestCase.java [hunk 3]
```diff
@@ -55,21 +56,15 @@ public void testSimpleCachingNotExists() throws AxisFault, XPathExpressionExcept
 
             response = axis2Client.sendSimpleStockQuoteRequest(getMainSequenceURL(), "", "IBM");
 
-            if (!firstResponse.equalsIgnoreCase(response.getFirstElement().toString())) {
-                assertFalse(false, "Caching is less than 20 seconds");
-            }
+            assertEquals(firstResponse, response.getFirstElement().toString(), "Caching is less than 20 seconds");
 
             Thread.sleep(2000);
```
Comment (Asitha) [inline integration/mediation-tests/tests-sample/src/test/java/org/wso2/carbon/esb/samples/test/mediation/cache/Sample420TestCase.java]: Why do we need a Thread.sleep(2000)?

Comment (riyafa) [reply integration/mediation-tests/tests-sample/src/test/java/org/wso2/carbon/esb/samples/test/mediation/cache/Sample420TestCase.java]: The timeout I have set in the Sample is 20s. The test here is to check if the cache is evicted after 20s. I have added a comment for this in PR #1167

**Pull request merged**